src/navigation/TabNavigator.tsx

```diff
@@ -9,42 +9,54 @@ import { HistoryNavigator } from './HistoryNavigator';
 import { HomeNavigator } from './HomeNavigator';
 import { ProfileNavigator } from './ProfileNavigator';
 
-export const TabNavigator = createBottomTabNavigator(
-  {
-    [Routes.HomeNavigator]: {
-      screen: HomeNavigator,
-      navigationOptions: {
-        tabBarIcon: ({ tintColor }) => <Ionicons name="ios-home" size={30} color={tintColor} />,
-      },
-    },
-    [Routes.WorkoutsNavigator]: {
-      screen: WorkoutsNavigator,
-      navigationOptions: {
-        tabBarIcon: ({ tintColor }) => <Ionicons name="ios-fitness" size={30} color={tintColor} />,
-      },
-    },
-    [Routes.HistoryNavigator]: {
-      screen: HistoryNavigator,
-      navigationOptions: {
-        tabBarIcon: ({ tintColor }) => <Ionicons name="ios-calendar" size={30} color={tintColor} />,
-      },
-    },
-    [Routes.ProfileNavigator]: {
-      screen: ProfileNavigator,
-      navigationOptions: {
-        tabBarIcon: ({ tintColor }) => <Ionicons name="ios-person" size={30} color={tintColor} />,
-      },
-    },
-  },
-  {
-    initialRouteName: Routes.HomeNavigator,
-    tabBarOptions: {
+export type TabNavigatorParamList = {
+  [Routes.HomeNavigator]: undefined;
+  [Routes.WorkoutsNavigator]: undefined;
+  [Routes.HistoryNavigator]: undefined;
+  [Routes.ProfileNavigator]: undefined;
+};
+
+const Tab = createBottomTabNavigator<TabNavigatorParamList>();
+
+export const TabNavigator = () => (
+  <Tab.Navigator
+    initialRouteName={Routes.HomeNavigator}
+    tabBarOptions={{
       showLabel: false,
       activeTintColor: colors.blue,
       inactiveTintColor: colors.white,
       style: {
         backgroundColor: colors.black,
       },
-    },
-  }
+    }}
+  >
+    <Tab.Screen
+      name={Routes.HomeNavigator}
+      component={HomeNavigator}
+      options={{
+        tabBarIcon: ({ color }) => <Ionicons name="ios-home" size={30} color={color} />,
+      }}
+    />
+    <Tab.Screen
+      name={Routes.WorkoutsNavigator}
+      component={WorkoutsNavigator}
+      options={{
+        tabBarIcon: ({ color }) => <Ionicons name="ios-fitness" size={30} color={color} />,
```

Comment:
extract icon size in a const

Reply:
ici tu pourrais faire un composant du genre:
```react
const TabBarIcon = (iconName) => ({color}) => <Ionicons name={iconName} size={30} color={color} />
```
et ensuite faire `tabBarIcon: TabBarIcon("ios-fitness")`